src/main/java/org/jenkinsci/plugins/deployment/DeploymentBuildWrapper.java

```diff
@@ -0,0 +1,51 @@
+package org.jenkinsci.plugins.deployment;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.Launcher;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.BuildListener;
+import hudson.tasks.BuildWrapper;
+import hudson.tasks.BuildWrapperDescriptor;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.util.Map;
+
+public class DeploymentBuildWrapper extends BuildWrapper {
```

Comment:
FYI, the selected approach won't work if SCM step fails. In such case variables won't be injected to the environment.

If it affects your use-cases, see the solution example in https://github.com/jenkinsci/ownership-plugin/pull/30